src/main.rs

```diff
@@ -46,47 +48,50 @@ enum Commands {
-                config.create_or_update_credentials(credentials).unwrap();
-            }
-            GlobalConfigCommands::Display => {
-                config.print();
+    Config::load()
+        .and_then(|mut config| match opts.command {
+            Commands::GlobalConfig { command } => match command {
+                GlobalConfigCommands::AddCredentials => {
+                    let credentials = ui::get_credentials();
+
+                    config
+                        .create_or_update_credentials(credentials)
+                        .and_then(|_| Ok(ExitCode::SUCCESS))
+                }
+                GlobalConfigCommands::Display => config.print().and_then(|_| Ok(ExitCode::SUCCESS)),
+            },
+            Commands::Init {} => {
+                println!("Init was called");
+                Ok(ExitCode::SUCCESS)
             }
-        },
-        Commands::Init {} => {
-            println!("Init was called");
-        }
-        Commands::New { path } => {
-            let credentials_url = &config
-                .get_credentials("https://jira.atlassian.com".to_string())
-                .unwrap()
-                .url;
-            let issues = drivers::mock::get_issues(credentials_url).unwrap_or_default();
+            Commands::New { path } => {
+                config
+                    .get_credentials("https://jira.atlassian.com".to_string())
+                    .map(|credentials| {
+                        let issues =
+                            drivers::mock::get_issues(&credentials.url).unwrap_or_default();
 
-            // TODO: Handle the branch name formatting
-            // let branch_name = format!(
-            //     "{}-{}",
-            //     issue_key,
-            //     jira::jira_get_issue_summary(&issue).replace(" ", "-")
-            // );
+                        // TODO: Handle the branch name formatting
+                        // let branch_name = format!(
+                        //     "{}-{}",
+                        //     issue_key,
+                        //     jira::jira_get_issue_summary(&issue).replace(" ", "-")
+                        // );
 
-            match fuzzy_finder::render(issues) {
-                Ok(issue) => {
-                    git::branch_create(path, issue.clone());
-                    println!("Created branch {}", issue);
-                }
-                Err(e) => {
-                    eprintln!("Error: {:?}", e);
-                    std::process::exit(1);
-                }
+                        match fuzzy_finder::render(issues) {
+                            Ok(issue) => {
+                                git::branch_create(path, issue.clone()).unwrap_or_else(|e| {
+                                    e.report();
+                                });
+                                ExitCode::SUCCESS
+                            }
+                            Err(e) => e.report(),
+                        }
```

Comment:
I couldn't find a better solution than handle the two `e.report()` together since the errors are from different types.